corehq/apps/es/management/commands/elastic_sync_multiplexed.py

```diff
@@ -267,33 +267,44 @@ def _get_current_and_older_index_name(cls, cname):
+    def set_checkpoints_for_new_index(self, cnames):
         """
-        Takes in an index cname and create new checkpoint for all the pillows that use the older index name
+        Takes in an index cnames and create new checkpoint for all the pillows that use the older index name
         for that cname.
+        If 'all' is passed, it will copy checkpoints for all the indices.
         Can only be performed when indexes are still multiplexed and not swapped.
         When we swap the indexes, the primary index changes which updates the checkpoint names.
         We should stop the pillows and copy the checkpoint to the new checkpoint ids, swap the indexes
         and then start the pillows.
         """
-        adapter = doc_adapter_from_cname(cname)
-        if not isinstance(adapter, ElasticMultiplexAdapter):
-            raise IndexNotMultiplexedException(f"""Checkpoints can be copied on multiplexed indexes.
-                Make sure you have set ES_{cname.upper()}_INDEX_MULTIPLEXED to True """)
-
-        current_index_name, older_index_name = self._get_current_and_older_index_name(cname)
-
-        if getattr(es_consts, f'ES_{cname.upper()}_INDEX_SWAPPED'):
-            raise IndexAlreadySwappedException(
-                f"""Checkpoints can only be copied before swapping indexes.
-                Make sure you have set ES_{cname.upper()}_INDEX_SWAPPED to False."""
-            )
-
         all_pillows = get_all_pillow_instances()
+        if 'all' in cnames:
+            cnames = iter_index_cnames()
+        for cname in cnames:
+            adapter = doc_adapter_from_cname(cname)
+            if not isinstance(adapter, ElasticMultiplexAdapter):
+                raise IndexNotMultiplexedException(f"""Checkpoints can be copied on multiplexed indexes.
+                    Make sure you have set ES_{cname.upper()}_INDEX_MULTIPLEXED to True """)
+
+            current_index_name, older_index_name = self._get_current_and_older_index_name(cname)
+
+            if getattr(es_consts, f'ES_{cname.upper()}_INDEX_SWAPPED'):
+                raise IndexAlreadySwappedException(
+                    f"""Checkpoints can only be copied before swapping indexes.
+                    Make sure you have set ES_{cname.upper()}_INDEX_SWAPPED to False."""
+                )
+
         for pillow in all_pillows:
             old_checkpoint_id = pillow.checkpoint.checkpoint_id
-            if older_index_name in old_checkpoint_id:
-                new_checkpoint_id = old_checkpoint_id.replace(older_index_name, current_index_name)
+            new_checkpoint_id = old_checkpoint_id
+            checkpoint_updated = False
+            for cname in cnames:
+                current_index_name, older_index_name = self._get_current_and_older_index_name(cname)
+                if older_index_name in old_checkpoint_id:
+                    new_checkpoint_id = new_checkpoint_id.replace(older_index_name, current_index_name)
```

Comment:
Is it possible for `new_checkpoint_id` to contain more than one occurrence of `older_index_name`? Should all of them be replaced?

Reply:
No it is not possible to have same index name twice in checkpoint name.